directory.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+#
+# Custom check for a directory presence and size (in Mbytes)
+#
+# if (metric['size'] < 0) {
+#   return new AlarmStatus(CRITICAL, 'Directory #{dir} check failed - no such directory?');
+# }
+# if (metric['size'] > 500) {
+#   return new AlarmStatus(WARNING, 'Directory #{dir} is #{size} Mbytes');
+# }
+# if (metric['size'] > 1000) {
+#   return new AlarmStatus(CRITICAL, 'Directory #{dir} is #{size} Mbytes');
+# }
+#
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 DIRECTORY"
+    exit 100
+fi
+
+DIR=$1
+
+size=-1
+if [ ! -d $DIR ]; then
+    status="err No such directory $DIR"
+else
+    status="ok directory $DIR found"
+    size=`du -sm $DIR 2>/dev/null | awk '{print $1}'`
+    if [ "X$size" = "X" ]; then
```

Comment:
I remember getting burned by the portability of this to something (solaris, AIX, ksh, ...?) a few times.  No absolute reference but http://unix.stackexchange.com/questions/32180/testing-if-a-variable-is-empty-in-a-shell-script points to a warning at http://resources.mpi-inf.mpg.de/departments/rg1/teaching/unixffb-ss98/quoting-guide.html#ssect:appl-test